src/ILLink.Shared/DataFlow/IDeepCopyValue.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace ILLink.Shared.DataFlow
+{
+	// Adds ability to deep copy a value
+	public interface IDeepCopyValue<TSingleValue>
```

Comment:
I think ideally this would be required by every type used as a value in a lattice, but we could do that in a separate change.

Reply:
Yeah - I think it makes sense to require it everywhere - this change does require it on SingleValue. Other values - I would also prefer a different PR>